### PR TITLE
Validate non-empty address in crosschain bridge transfers

### PR DESCRIPTION
**Pull request by Amxx**
## Motivation

The abstract crosschain bridges (`BridgeFungible`, `BridgeNonFungible`, `BridgeMultiToken`) parse the destination as an ERC-7930 interoperable address and forward its address part in the ERC-7786 payload. If the address part is empty, the transfer would be sent with no recipient on the destination chain, silently locking the tokens.

## Changes

- Add an `Crosschain*EmptyAddress` custom error to each of the three abstract bridges and `require` that the parsed address part is non-empty before dispatching the crosschain message.
- Add tests covering the empty-address case for ERC-20, ERC-721 and ERC-1155 bridges.
- Allow `chains.toErc7930(undefined)` in the test helper so the empty-address case can be constructed.

## Notes

No changeset: these contracts are unreleased (`contracts/crosschain/bridges/`), so there is no user-visible behavior change to a shipped API.

### CODE REVIEW

File: contracts/crosschain/bridges/abstract/BridgeNonFungible.sol [hunk 1]
```diff
@@ -29,6 +29,9 @@ abstract contract BridgeNonFungible is Context, CrosschainLinked {
         uint256 tokenId
     );
 
+    /// @dev Revert reason when the address part of the interoperable address is empty.
+    error CrosschainNonFungibleEmptyAddress();
+
     /**
      * @dev Internal crosschain transfer function.
```
[the remark this answers is not in the document]
Comment (james-toussaint) [reply contracts/crosschain/bridges/abstract/BridgeNonFungible.sol:36]: Receiver part is made for evm address length 20. Send part is length agnostic hence only > 0 check.

File: contracts/crosschain/bridges/abstract/BridgeMultiToken.sol [hunk 2]
```diff
@@ -40,6 +40,9 @@ abstract contract BridgeMultiToken is Context, CrosschainLinked {
         bytes data
     );
 
+    /// @dev Revert reason when the address part of the interoperable address is empty.
+    error CrosschainMultiTokenEmptyAddress();
+
     /**
      * @dev Internal crosschain transfer function. `data` is forwarded through the ERC-7786 payload to
```
[the remark this answers is not in the document]
Comment (james-toussaint) [reply contracts/crosschain/bridges/abstract/BridgeMultiToken.sol:47]: https://github.com/OpenZeppelin/openzeppelin-contracts/pull/6654#discussion_r3664072252

File: contracts/crosschain/bridges/abstract/BridgeFungible.sol [hunk 3]
```diff
@@ -26,6 +26,9 @@ abstract contract BridgeFungible is Context, CrosschainLinked {
     /// @dev Emitted when a crosschain ERC-20 transfer is received.
     event CrosschainFungibleTransferReceived(bytes32 indexed receiveId, bytes from, address indexed to, uint256 amount);
 
+    /// @dev Revert reason when the address part of the interoperable address is empty.
+    error CrosschainFungibleEmptyAddress();
+
     /**
      * @dev Transfer `amount` tokens to a crosschain receiver.
```
[the remark this answers is not in the document]
Comment (james-toussaint) [reply contracts/crosschain/bridges/abstract/BridgeFungible.sol:33]: https://github.com/OpenZeppelin/openzeppelin-contracts/pull/6654#discussion_r3664072252

**Review by james-toussaint — APPROVED** [review]
